pkg/controllers/utils/common_test.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2020 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util/gclient"
+)
+
```

Comment:
This test fails with:
```
E0722 13:25:50.730922   76848 common.go:27] Invalid failover type passed into updateFailoverStatus
--- FAIL: TestUpdateFailoverStatus (0.00s)
    --- FAIL: TestUpdateFailoverStatus/application_failover (0.00s)
        common_test.go:69: 
                Error Trace:    /Users/adityaaddepalli/Documents/karmada/pkg/controllers/utils/common_test.go:69
                Error:          Received unexpected error:
                                resourcebindings.work.karmada.io "binding" not found
                Test:           TestUpdateFailoverStatus/application_failover
    --- FAIL: TestUpdateFailoverStatus/cluster_failover (0.00s)
        common_test.go:69: 
                Error Trace:    /Users/adityaaddepalli/Documents/karmada/pkg/controllers/utils/common_test.go:69
                Error:          Received unexpected error:
                                resourcebindings.work.karmada.io "binding" not found
                Test:           TestUpdateFailoverStatus/cluster_failover
FAIL
FAIL    github.com/karmada-io/karmada/pkg/controllers/utils     3.257s
```
I think this is due to the mocked client unable to perform actions on the resource binding.
 
UpdateFailoverStatus calls [UpdateStatus](https://github.com/karmada-io/karmada/blob/bf2a6044a0de3a6219933dbd029e005cfaf7569c/pkg/util/helper/status.go#L39) which performs a get and mutate on the resouce. I think this error is coming from the get call in that function.

Is there is a way to get around this, I don't see test functions for other methods like [AggregateResourceBindingWorkStatus](https://github.com/karmada-io/karmada/blob/bf2a6044a0de3a6219933dbd029e005cfaf7569c/pkg/util/helper/workstatus.go#L58) in the respective [test file ](https://github.com/karmada-io/karmada/blob/bf2a6044a0de3a6219933dbd029e005cfaf7569c/pkg/util/helper/workstatus_test.go)either 

Would I need to write my own on mocked UpdateStatus for this?